src/turner_lab_to_nwb/asap_tdt/asap_tdt_convert_session.py

```diff
@@ -1,204 +1,203 @@
-    if location is not None:
-        assert location in ["GPi", "VL"], f"Location must be one of ['GPi', 'VL'], not {location}."
+    channel_metadata = session_metadata.to_dict(orient="list")
 
-    # Add Recording
-    recording_source_data = dict(
-        file_path=str(ecephys_file_path), data_list_file_path=str(data_list_file_path), location=location, gain=1.0
-    )
-    source_data.update(dict(Recording=recording_source_data))
-    conversion_options.update(dict(Recording=dict(stub_test=stub_test)))
-
-    # Add Sorting (uncurated spike times from Plexon Offline Sorter v3)
-    conversion_options_sorting = dict(
-        stub_test=stub_test,
-        write_as="processing",
-        units_description="The units were sorted using the Plexon Offline Sorter v3.",
-    )
-
-    # Add Processed Recording (high-pass filtered data)
-    if vl_flt_file_path:
-        source_data.update(dict(ProcessedRecordingVL=dict(file_path=str(vl_flt_file_path), location="VL")))
-        conversion_options.update(dict(ProcessedRecordingVL=dict(stub_test=stub_test, write_as="processed")))
-    if gpi_flt_file_path:
-        source_data.update(dict(ProcessedRecordingGPi=dict(file_path=str(gpi_flt_file_path), location="GPi")))
-        conversion_options.update(dict(ProcessedRecordingGPi=dict(stub_test=stub_test, write_as="processed")))
-
-    # Add Sorting (uncurated spike times from Plexon Offline Sorter v3)
-    if vl_plexon_file_path:
-        source_data.update(dict(PlexonSortingVL=dict(file_path=str(vl_plexon_file_path))))
-        conversion_options.update(dict(PlexonSortingVL=conversion_options_sorting))
-    if gpi_plexon_file_path:
-        source_data.update(dict(PlexonSortingGPi=dict(file_path=str(gpi_plexon_file_path))))
-        conversion_options.update(dict(PlexonSortingGPi=conversion_options_sorting))
-
-    # Add Sorting (curated spike times from Plexon Offline Sorter v3, only include single units)
-    source_data.update(dict(CuratedSorting=dict(file_path=str(events_file_path), location=location)))
-    conversion_options.update(
-        dict(
+    # Temporary until TDT files are fixed that have missing "StoreName" header key error.
+    try:
+        recording_interface = ASAPTdtRecordingInterface(
+            file_path=str(tdt_tank_file_path),
+            channel_metadata=channel_metadata,
+            gain=1.0,
+        )
+    except Exception as e:
+        print(f"Error in recording interface for session {tdt_tank_file_path}: {e}")
+        return
+
+    data_interfaces.update(Recording=recording_interface)
+    conversion_options.update(Recording=dict(stub_test=stub_test))
+
+    events_interface = ASAPTdtEventsInterface(file_path=events_file_path)
+    data_interfaces.update(Events=events_interface)
+    conversion_options.update(Events=dict(target_name_mapping=target_name_mapping))
+
+    # Check if the session is GPI only
+    gpi_only = any(["GP" in target for target in session_metadata["Target"].values])
```

Comment:
Question: why are you checking if at least one is GP? Shouldn't be all GP to be a gpi_only session?

Reply:
Here it's basically just a flag, maybe passing it as a kwarg in `convert_session.py` would be more readable? The main "filtering" happens after e.g. in filtering units 
```
units_df[units_df["brain_area"].str.contains("GP") == gpi_only]
```
I can add this small change (to just pass it asa function argument) in #17